docs/route.md

```diff
@@ -490,9 +490,13 @@ In example 2, since `Layout` is the top-level component in both routes, the DOM
 
 ---
 
-#### Authentication
+#### Redirection
 
-The RouterResolver's `onmatch` hook can be used to run logic before the top level component in a route is initializated. The example below shows how to implement a login wall that prevents users from seeing the `/secret` page unless they login.
+The RouterResolver's `onmatch` hook can be used to run logic before the top level component in a route is initializated. You can use either `m.route.set()` or `window.history.back()` `.forward()`, etc... When redirecting with the `history` API, the `onmatch` hook must return `new Promise(function(){})` to suspend the resolution of the current route. A fresh Promise must be returned every time since recycling a pending Promise leads to memory leaks. `m.route.set()` takes care of cancelling the resolution of the aborted route, there's no need to return a pending Promise in that case.
```

Comment:
```suggestion
The RouteResolver's `onmatch` hook can be used to run logic before the top level component in a route is initialized. You can use either Mithril's `m.route.set()` or native HTML's `history` API. When redirecting with the `history` API, the `onmatch` hook must return a never-resolving Promise to prevent resolution of the matched route. `m.route.set()` cancels resolution of the matched route internally, so this isn't necessary with it.
```

I'll take this sentence by sentence:

- First, two minor nits with the first sentence: `RouteResolver` and `initialized`, not `RouterResolver` and `initializated`. I know this was incorrect in the original docs, but we should probably correct them here while we're at it.
- The second sentence is missing a comma and using an unnecessary ellipsis, but instead of correctng it, we should reword it to just refer to functionality, not specific methods. By doing so, we not only avoid the Latin-based "etc." that frequently confuses less-experienced non-native speakers, but we also make it easier to read and understand overall.
- Changed the third sentence to just say "never-resolving Promise" to clarify the intent. Anyone familiar with JS promises should know that this means use `new Promise(() => {})` or similar. Also, this gets rid of some ES5 syntax crud.
- Let's just strike "A fresh Promise must be returned every time since recycling a pending Promise leads to memory leaks." for now. https://bugs.chromium.org/p/v8/issues/detail?id=6197 can't find anything going on, and unless we gain new hard evidence down that vein that we can actually reference (like 
a link or example command), it's just going to mislead the reader.
- The last sentence is a [run-on](https://academicguides.waldenu.edu/writingcenter/grammar/runonsentences), but I've recast it to be a lot more concise and flow from the third sentence. (Removing the fourth made the recast possible, actually.)